src/Microsoft.DotNet.ImageBuilder/src/Commands/CleanAcrImagesCommand.cs

```diff
@@ -109,15 +109,21 @@ private async Task LogSummaryAsync(IAcrClient acrClient, List<string> deletedRep
         }
 
         private async Task ProcessTestRepoAsync(
-            IAcrClient acrClient, List<string> deletedImages, Repository repository)
+            IAcrClient acrClient, List<string> deletedImages, List<string> deletedRepos, Repository repository)
         {
             RepositoryManifests repoManifests = await acrClient.GetRepositoryManifestsAsync(repository.Name);
+            if (!repoManifests.Manifests.Any())
```

Comment:
Is it possible to incorporate the expired check so that
1. the repo is deleted if all manifests are expired
1. only one operation is necessary

```suggestion

            IEnumerable<ManifestAttributes> expiredTestImages = repoManifests.Manifests
                .Where(manifest => IsExpired(manifest.LastUpdateTime, 7));
                .ToArray();
            if (expiredTestImages.Count() == repoManifests.Manifests.Count())
            {
                await DeleteRepositoryAsync(acrClient, deletedRepos, repository);
                return;
            }
            
            await DeleteManifestsAsync(acrClient, deletedImages, repository, expiredTestImages);
```

Reply:
What do you mean by "only one operation is necessary"?  One operation to do what?

Reply:
I mean if the repository only contains expired manifests, all expired manifests and the repository itself can be deleted with one request.

Reply:
Great suggestion.